crates/edr_rpc_server/src/node.rs

```diff
@@ -390,6 +398,169 @@ impl Node {
 
         node_data.impersonated_accounts.remove(&address)
     }
+
+    pub async fn block_and_total_difficulty_by_block_spec(
+        &self,
+        block_spec: &BlockSpec,
+    ) -> Result<BlockAndTotalDifficulty, NodeError> {
+        let mut node_data = self.lock_data().await;
+
+        let block = node_data.block_by_block_spec(block_spec).await?;
+
+        let total_difficulty = node_data
+            .blockchain
+            .total_difficulty_by_hash(block.hash())
+            .await?
+            .expect("The block exists, so it must have a total difficulty");
+
+        self.workaround_block_by_spec(&mut *node_data.blockchain, block_spec)
+            .await?;
+
+        Ok(BlockAndTotalDifficulty {
```

Comment:
Is this notation idiomatic? I mean `Ok(LiteralConstructor { ... })`, with that level of indentation.

Reply:
Yes 👍